billing/customer/service_test.go

```diff
@@ -0,0 +1,371 @@
+			want: customer.Customer{
+				ID:    "1",
+				Name:  "updated_customer1",
+				OrgID: "org1",
+			},
+			wantErr: nil,
+			setup: func() *customer.Service {
+				stripeClient, mockStripeBackend, mockRepo := mockService(t)
+
+				mockRepo.EXPECT().GetByID(ctx, "1").Return(customer.Customer{
+					ID:    "1",
+					Name:  "customer1",
+					OrgID: "org1",
+				}, nil)
+
+				mockStripeBackend.EXPECT().Call("POST", "/v1/customers/", "key_123",
+					&stripe.CustomerParams{
+						Params: stripe.Params{
+							Context: ctx,
+						},
+						Address: &stripe.AddressParams{
+							City:       stripe.String(""),
+							Country:    stripe.String(""),
+							Line1:      stripe.String(""),
+							Line2:      stripe.String(""),
+							PostalCode: stripe.String(""),
+							State:      stripe.String(""),
+						},
+						Email:     stripe.String(""),
+						Name:      stripe.String("customer1"),
+						Phone:     stripe.String(""),
+						TaxIDData: nil,
+						Metadata: map[string]string{
+							"managed_by": "frontier",
+							"org_id":     "org1",
+						},
+					}, &stripe.Customer{ID: ""}).Return(nil)
+
+				mockRepo.EXPECT().
+					UpdateByID(ctx, customer.Customer{
+						ID:    "1",
+						Name:  "customer1",
+						OrgID: "org1"}).
+					Return(customer.Customer{
+						ID:    "1",
+						Name:  "updated_customer1",
+						OrgID: "org1",
+					}, nil)
+
+				cfg := billing.Config{}
+
+				return customer.NewService(stripeClient, mockRepo, cfg)
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := tt.setup()
+			got, err := s.Update(ctx, tt.args.customer)
+
+			if tt.wantErr != nil {
+				require.Error(t, err)
+				assert.ErrorIs(t, err, tt.wantErr)
+			} else {
+				require.NoError(t, err)
+			}
+
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestService_GetByID(t *testing.T) {
+	ctx := context.Background()
+	type args struct {
+		id string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    customer.Customer
+		wantErr error
+		setup   func() *customer.Service
+	}{
+		{
+			name: "successfully get customer by id",
+			args: args{
+				id: "1",
+			},
+			want: customer.Customer{
+				ID:    "1",
+				Name:  "customer1",
+				OrgID: "org1",
+			},
+			wantErr: nil,
+			setup: func() *customer.Service {
+				stripeClient, _, mockRepo := mockService(t)
+				mockRepo.EXPECT().GetByID(ctx, "1").Return(
+					customer.Customer{
+						ID:    "1",
+						Name:  "customer1",
+						OrgID: "org1",
+					}, nil)
+
+				cfg := billing.Config{}
+
+				return customer.NewService(stripeClient, mockRepo, cfg)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := tt.setup()
+			got, err := s.GetByID(ctx, tt.args.id)
+			if tt.wantErr != nil {
+				require.Error(t, err)
+				assert.ErrorIs(t, err, tt.wantErr)
+			} else {
+				require.NoError(t, err)
+			}
+
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestService_List(t *testing.T) {
```

Comment:
Same here.
I understand that we're only calling the repository and returning its error, but the unit tests will ensure that any time someone changes that implementation, they are aware that the contract and behaviour have changed.